sharedbuffers/mapped_struct.py

```diff
@@ -1547,6 +1588,22 @@ def __init__(self, buf, offset = 0, idmap = None, idmap_size = 1024):
             count = self.index_elements)
         self.idmap = idmap
 
+        if self.index_elements > 0 and self.index[0] >= (self._Header.size + self._NewHeader.size):
+            # New version, most likely
+            self.version, self.schema_offset, self.schema_size = self._NewHeader.unpack_from(buf, self._Header.size)
+            if self.schema_offset and self.schema_size:
+                if self.schema_offset > len(buf) or (self.schema_size + self.schema_offset) > len(buf):
+                    raise ValueError("Corrupted input - bad schema location")
+                stored_schema = cPickle.loads(bytes(buffer(buf, self.schema_offset, self.schema_size)))
+                if not isinstance(stored_schema, Schema):
+                    raise ValueError("Corrupted input - unrecognizable schema")
+                if self.schema is None or not self.schema.compatible(stored_schema):
+                    self.schema = stored_schema
+            elif self.schema is None:
+                raise ValueError("Cannot map schema-less buffer without specifying schema")
+        else:
+            self.version = 0
```

Comment:
you mean know it from previous versions? ... how would that work here? b/c that makes sense for logging extra values but I'm not sure there's much value in keeping unused code here.